### PR TITLE
[Animal][AnimalGenetics] AnimalGenetics Setup [1/2]

### LINKED ISSUE #1695: [Animal][AnimalGenetic] Genetic Setup
Every animal simulated in the RuFaS model should have an associated genetic value. We choose to use the net-merit value.

Create a new `genetic_value`/`net_merit_value` property in the `Animal` class to represent the genetic/net-merit value of the animal.

Follow the instructions in the [AnimalPrescibeFix](https://docs.google.com/document/d/1f231Koa7dBlsBTAXD6L1qBhZ_F6ndl9uy63CSeud4h8/edit#heading=h.lf57q7ulfzse) document to set up the required CSV documents.
1. Create the documents following instructions. If you have any issues, reach out to @YijingGong for more details.
2. Modify the metadata to include the new inputs.

In the `Animal` class, create a new @property that returns the animal's birth year (in calendar year) based on the current time and the `days_born` value.

Create a class `AnimalGenetic` to handle genetic-related calculations. This class should have the following methods:
1. During initialization, load the necessary input data from the `InputManager`
2. Assign genetic value:
    1. Assign a genetic value for animals loaded from the input JSON file.
    2. Assign a genetic value for animals entering the herd (replacement animals).
    3. Calculate the genetic value for newborn calves during the simulation.
        1. A function to calculate the standard deviation using [Mendelian sampling variance](https://gsejournal.biomedcentral.com/articles/10.1186/s12711-016-0214-0). You don't have to implement the logic; create the instruction with a placeholder here.
2. Report genetic value:
    1. In `HerdManager.daily_updates()`, report the daily average genetic value of the herd to the OutputManager

**Target Deadline**
By when should this issue be done.

### PR DESCRIPTION
**Pull request by allisterakun**
This PR set up the AnimalGenetics submodule.
### Context

Issue(s) closed by this pull request: closes #1695 

- [ ] The [changelog](https://docs.google.com/spreadsheets/d/1U-3igxdstHTFb6k5dVcE5-x9IL0r6q18eUiwwytKZyA/edit#gid=0) has been updated.
      
How big are the changes in the PR? Would you nominate it for a major version upgrade?
- [x] Yes
- [ ] No

### What

- Set up the `AnimalGenetics` class with the following methods:
    - `__init__()` to retrieve the net merit percentile values and the top listing semen values from the InputManager upon initialization
    - `assign_net_merit_value_to_animals_entering_herd()` to assign a net merit value to animals entering the herd: either during initialization or for animals bought during the simulation.
    - `assign_net_merit_value_to_newborn_calf()` to assign a net merit value to the animals born during the simulation.
- Created two CSVs:
    - `NetMerit_HO.csv` contains the percentile net merit values from 2006 to 2024 for the Holstein cows.
    - `TopListingSement_HO.csv` contains the net merit values for the top listing semen from 2006 to 2024 for the Holstein cows.

### Why

To set up the sub-module structure for the AnimalGenetics.

### How

For `TopListingSement_HO.csv`, I retrieved 17 available data points on the CDCB website (https://webconnect.uscdcb.com/#/top-animal-listing). I created a line of best fit from these data points to estimate the missing values. This can be found in [TopListingSemen](https://docs.google.com/spreadsheets/d/1T0SPjUQ1jAWafi3XkbAFzo0c32Zk3mvYv56TvbhA7wY/edit?usp=sharing).

For `NetMerit_HO.csv`, @YijingGong provided Holstein's percentile net merit values from 2006 to 2024. These data are provided in HTML files, and I created the following script to iteratively go through all HTML files to extract the data and put together a CSV of the net merit values for Holstein cows.
```python
import os

import pandas as pd
from bs4 import BeautifulSoup

html_dir = "DATA_REQUEST_20231006_unzipped/"
html_files = os.listdir(html_dir)
html_files = sorted(html_files)

output_path = html_dir + "NetMerit_HO.csv"

output_df = pd.DataFrame(index=list(reversed(range(101))))

for html_file in html_files:
    year_month = html_file.split(".")[0][-4:]
    year_month = "20" + year_month[:2] + "-" + year_month[2:]

    values = []
    with open(html_dir + html_file) as html_fp:
        soup = BeautifulSoup(html_fp, 'html.parser')
        table_body = soup.find('tbody')
        table_rows = table_body.find_all('tr')
        for tr in table_rows:
            table_data = tr.find_all('td')
            tens = table_data[0].contents[0].strip()
            table_data = table_data[1:]
            for i in reversed(range(len(table_data))):
                percentile = int(tens) + i
                if percentile > 100:
                    continue
                value = int(table_data[i].contents[0].strip())
                values.append(value)
    output_df.insert(len(output_df.columns), year_month, values)

output_df.to_csv(output_path)

```
### Test plan

No tests are needed as this task is only for setting up the class structure for AnimalGenetics submodule.

### CODE REVIEW

File: tests/test_task_manager.py
Comment (allisterakun) [inline tests/test_task_manager.py:1]: I did not make any changes to `test_task_manager.py`; not sure how these changes showed up in this PR.

File: RUFAS/routines/animal/genetics/__init__.py
Comment (ew3361zh) [inline RUFAS/routines/animal/genetics/__init__.py:1]: Curious why you're including the empty init file? I don't believe these are widely used anymore with newer Python versions and I haven't seen us doing this across other packages. Is the idea to have one for each new animal submodule?

**Review by ew3361zh — APPROVED** [review]
LGTM!

File: tests/test_task_manager.py
Comment (ew3361zh) [reply tests/test_task_manager.py:1]: Have you tried the trick Ed mentioned where you rebase this and then return it to base off dev (using the edit at the top of the PR)?
Screenshot 2024-08-14 at 9 33 12 AM

File: RUFAS/routines/animal/genetics/__init__.py
Comment (allisterakun) [reply RUFAS/routines/animal/genetics/__init__.py:1]: I was using Pycharm to create the folder, and I think I selected the option for a package, so this __init__.py is auto-generated in the process. I will remove this!

File: tests/test_task_manager.py
Comment (allisterakun) [reply tests/test_task_manager.py:1]: Just did, and it works!

**Review by YijingGong — APPROVED** [review]
Perfect work! I randomly checked some values of $NM of cows and they match exactly! Two follow-up function requests would be 1) genetic base changes need to be incorporated [we can schedule a meeting about it] 2) include NM for cows from 2022 to 2024 from CDCB website (https://webconnect.uscdcb.com/#/summary-stats/merit-percentile)

**Pull request merged**